integrationTests/chainSimulator/common.go

```diff
@@ -112,30 +114,99 @@ func SendTransaction(
+	sender []byte,
+	nonce *uint64,
+	issueCost *big.Int,
+	esdtName string,
+	esdtTicker string,
+	tokenType string,
+	numDecimals int,
+) string {
+	esdtType := getTokenRegisterType(tokenType)
+	registerArgs := "registerAndSetAllRoles" +
+		"@" + hex.EncodeToString([]byte(esdtName)) +
+		"@" + hex.EncodeToString([]byte(esdtTicker)) +
+		"@" + hex.EncodeToString([]byte(esdtType)) +
+		"@" + fmt.Sprintf("%02X", numDecimals)
+	SendTransaction(t, cs, sender, nonce, vm.ESDTSCAddress, issueCost, registerArgs, uint64(60000000))
+
+	return getEsdtIdentifier(t, nodeHandler, esdtTicker, tokenType)
+}
+
+func getTokenRegisterType(tokenType string) string {
+	switch tokenType {
+	case core.FungibleESDT:
+		return "FNG"
+	case core.NonFungibleESDT:
+		return "NFT"
+	case core.SemiFungibleESDT:
+		return "SFT"
+	case core.MetaESDT:
+		return "METTA"
+	}
+	return ""
+}
+
 // IssueFungible will issue a fungible token
 func IssueFungible(
 	t *testing.T,
 	cs ChainSimulator,
+	nodeHandler process.NodeHandler,
 	sender []byte,
 	nonce *uint64,
 	issueCost *big.Int,
 	tokenName string,
 	tokenTicker string,
 	numDecimals int,
 	supply *big.Int,
-) []byte {
+) string {
 	issueArgs := "issue" +
 		"@" + hex.EncodeToString([]byte(tokenName)) +
 		"@" + hex.EncodeToString([]byte(tokenTicker)) +
 		"@" + hex.EncodeToString(supply.Bytes()) +
 		"@" + fmt.Sprintf("%X", numDecimals) +
 		"@" + hex.EncodeToString([]byte("canAddSpecialRoles")) +
 		"@" + hex.EncodeToString([]byte("true"))
-	txResult := SendTransaction(t, cs, sender, nonce, vm.ESDTSCAddress, issueCost, issueArgs, uint64(60000000))
-	tokenIdentifier := txResult.Logs.Events[0].Topics[0]
-	require.Equal(t, len(tokenTicker)+7, len(tokenIdentifier))
-	require.Equal(t, tokenName, string(txResult.Logs.Events[4].Topics[1]))
-	require.Equal(t, tokenTicker, string(txResult.Logs.Events[4].Topics[2]))
+	SendTransaction(t, cs, sender, nonce, vm.ESDTSCAddress, issueCost, issueArgs, uint64(60000000))
+
+	return getEsdtIdentifier(t, nodeHandler, tokenTicker, core.FungibleESDT)
+}
+
+// IssueSemiFungible will issue a semi fungible token
+func IssueSemiFungible(
+	t *testing.T,
+	cs ChainSimulator,
+	nodeHandler process.NodeHandler,
+	sender []byte,
+	nonce *uint64,
+	issueCost *big.Int,
+	sftName string,
+	sftTicker string,
+) string {
+	issueArgs := "issueSemiFungible" +
+		"@" + hex.EncodeToString([]byte(sftName)) +
+		"@" + hex.EncodeToString([]byte(sftTicker))
+	SendTransaction(t, cs, sender, nonce, vm.ESDTSCAddress, issueCost, issueArgs, uint64(60000000))
+
+	return getEsdtIdentifier(t, nodeHandler, sftTicker, core.SemiFungibleESDT)
+}
+
+func getEsdtIdentifier(t *testing.T, nodeHandler process.NodeHandler, ticker string, tokenType string) string {
+	issuedTokens, err := nodeHandler.GetFacadeHandler().GetAllIssuedESDTs(tokenType)
+	require.Nil(t, err)
+	require.GreaterOrEqual(t, len(issuedTokens), 1)
+
+	for _, issuedToken := range issuedTokens {
+		if strings.Contains(issuedToken, ticker) {
```

Comment:
This will only work as long as we don't issue two tokens with the same ticker :D.
It's good enough for now

Reply:
Yes, saw that.